pkg/network/dns/driver_windows.go

```diff
@@ -61,18 +61,38 @@ func (d *dnsDriver) setupDNSHandle() error {
 		return err
 	}
 
-	if err := dh.SetDataFilters(filters); err != nil {
+	// must assign handle value before calling SetDataFilters because SetDataFilters
+	// uses it.
+	d.h = dh
+	if err := d.SetDataFilters(filters); err != nil {
 		return err
 	}
 
-	iocp, buffers, err := prepareCompletionBuffers(dh.Handle, dnsReadBufferCount)
+	iocp, buffers, err := prepareCompletionBuffers(dh.GetWindowsHandle(), dnsReadBufferCount)
 	if err != nil {
 		return err
 	}
 
 	d.iocp = iocp
 	d.readBuffers = buffers
-	d.h = dh
+
+	return nil
+}
+
+// SetDataFilters installs the provided filters for data
+func (d *dnsDriver) SetDataFilters(filters []driver.FilterDefinition) error {
```

Comment:
This should also be on the `Handle` interface?

Reply:
Thank you for the suggestion. As discussed in Slack, will move to different PR as needed.